### DISCUSSION TITLE
probably ROADMAP 🚀

### DISCUSSION THREAD

**Discussion by U-C-S** [U-C-S/Hurl#156] [General]
As of v0.6.3, For me, the Hurl is almost fitting for all of my use cases. Additional features below might make this project more useful for any future users:

- [x] Proper Communication with Hurl Browser Extension 
- [x] Context menu option `Open this browser for next 10mins automatically`
- [ ] Better Icon retrieval methods (currently, Icons are taken at runtime from browser exe files)
- [x] Hotkeys Support (Partially)
- [x] Hurl Settings GUI
- [ ] Fetching Browser profiles automatically
- [ ] History upto 10 items
- [x] #18

I intend to work on those slowly maybe like spending a week per month or randomly.

EDIT (2023-02-10): About the Code Quality, I feel like It's a absolute mess. Inventing patterns, Frequent Refactors all over, reverting/reusing old good ideas..... I can only think of fixing all slowly as I learn and experience more of the language and framework. Maybe look about it for stable release.

EDIT (2024-03-20): Code quality for the newer Ruleset Manager is much better, Looking to Refactor Browser Selector now with similar patterns.

**Comment by mix3d** [comment]
That last one, automatic rules, is what I was looking for in a Windows alternative to Velja or Choosey. Anything I can help with there?

**Comment by riotrah** [comment]
Echoing @mix3d, I also came here looking for Choosey, browserosaurus and Velja windows alternatives. 

I'm trying to port browserosaurus to windows but I have literally no clue how to enable an (electron) app to register itself as a browser, so I'd be happy to divert time to this instead, though I have no C# experience, and my last time doing OOP was like ~2 years ago. Still, this project is so promising and great already that I'm very excited about it. Thanks @U-C-S !

**Comment by U-C-S** [comment]
Thanks ☺️.
